src/core_modules/capture-core/components/D2Form/FormFieldPlugin/FormFieldPlugin.container.js

```diff
@@ -6,11 +6,13 @@ import { usePluginMessages } from './hooks/usePluginMessages';
 import { usePluginCallbacks } from './hooks/usePluginCallbacks';
 import { usePluginValues } from './hooks/usePluginValues';
 import { formatPluginConfig } from './formatPluginConfig';
+import { useLocationQuery } from '../../../utils/routing';
 
 export const FormFieldPlugin = (props: ContainerProps) => {
     const { pluginSource, fieldsMetadata, formId, onUpdateField, pluginContext } = props;
     const metadataByPluginId = useMemo(() => Object.fromEntries(fieldsMetadata), [fieldsMetadata]);
     const configuredPluginIds = useMemo(() => Object.keys(metadataByPluginId), [metadataByPluginId]);
+    const { programId, orgUnitId } = useLocationQuery();
```

Comment:
Hey,
What about cases where `programId` is missing from URL parameters (such as on `/viewEvent` or `/enrollmentEventEdit` pages)? Should we take it from somewhere else in these cases? 🤔 
Thanks!

Reply:
That's a good point. Also, I didn't think this quite through I guess. Instead of sending in programId, we should perhaps send in the contextId, i.e. either programId or TETid? Also, they configure the plugin by this contextId, so it should be familiar. WDYT?

Reply:
Do they need to know if the plugin is injected within a program or a TET context? If not, then a generic `contextId` makes sense to me.